File: tier1/go/ec6f4d6e/reader.go
package app

import (
	"errors"
	"os"
	"path/filepath"
	"strings"
)

const base = "/srv/reports"

func Contents(name string) ([]byte, error) {
	target := filepath.Join(base, filepath.Base(name))
	if !strings.HasPrefix(target, base+string(filepath.Separator)) {
		return nil, errors.New("rejected")
	}
	return os.ReadFile(target)
}

File: tier1/rust/e973850c/reader.rs
use std::fs;
use std::path::Path;

const BASE: &str = "/srv/reports";

pub fn contents(name: &str) -> String {
    let leaf = match Path::new(name).file_name() {
        Some(value) => value.to_owned(),
        None => return String::new(),
    };
    let target = Path::new(BASE).join(leaf);
    fs::read_to_string(target).unwrap_or_default()
}
